win_etw_macros/src/lib.rs

```diff
@@ -1173,33 +1187,43 @@ impl syn::parse::Parse for ProviderAttributes {
             });
         }
 
-        // We could generate a deterministic GUID by hashing the event provider name or the
-        // signatures of the event methods. Both of those approaches have problems, unfortunately.
-        // It's best to require developers to specify a GUID.
-        //
-        // I considered generating a GUID and printing it in the error message, but I decided not
-        // to do that because it introduces non-determinism.
-        let uuid = if let Some(provider_uuid) = uuid_opt {
-            provider_uuid
-        } else {
-            errors.push(Error::new_spanned(
-                &items,
-                "The 'guid' attribute is required.
-Please generate a GUID that uniquely identfies this event provider.
-Do not use the same GUID for different event providers.
-Example: #[trace_logging_provider(guid = \"123e4567-e89b...\")]",
-            ));
-            Uuid::nil()
-        };
-
         errors.into_result(ProviderAttributes {
-            uuid,
+            uuid: uuid_opt,
             provider_name,
             provider_group_guid,
         })
     }
 }
 
+const ETW_EVENT_SOURCE_NAMESPACE: Uuid = uuid::uuid!("482c2db2-c390-47c8-87f8-1a15bfc130fb");
+
+/// Generates a Uuid from a provider name using the same algorithm as .NET's EventSource class.
+/// Many tools convert a provider name to a GUID using this algorithm.
+fn etw_event_source_guid(provider_name: &str) -> Uuid {
+    use sha1_smol::Sha1;
+
+    let provider_bytes: Vec<u8> = provider_name
+        .to_uppercase()
+        .encode_utf16()
+        .flat_map(|x| x.to_be_bytes())
+        .collect();
+
+    let mut hasher = Sha1::new();
```

Comment:
Hmmm.  I understand that this is correct and does the same thing as .NET, but this is going to trigger some alarm bells in our secure supply-chain system, because SHA1 is considered compromised.  Let's go ahead with this, but it's going to cause a little grief on my side.

Reply:
Hopefully, there's an easy way to suppress that.